tests/codemods/tests/index.spec.ts

```diff
@@ -0,0 +1,168 @@
+  dirName: string,
+  transformName: string,
+  fixturesPath: string,
+  options?: Options | null,
+  testOptions?: TestOptions
+): void {
+  const testFilePrefix = `${transformName}/${fixturesPath}`;
+  const testName = testFilePrefix ? `transforms correctly using "${testFilePrefix}" data` : 'transforms correctly';
+  describe(transformName, () => {
+    it(testName, async () => {
+      await runTest(dirName, transformName, fixturesPath, options, testOptions);
+    });
+  });
+}
+
+async function runTest(
+  dirName: string,
+  transformName: string,
+  fixturesPath: string,
+  options?: Options | null,
+  testOptions: TestOptions = {}
+): Promise<void> {
+  const realLoggerWarn = console.warn;
+  const logs: Array<unknown[]> = [];
+  console.warn = (...args) => {
+    logs.push(['warn', ...args]);
+  };
+
+  const testFilePrefix = `${transformName}/${fixturesPath}`;
+
+  // Assumes transform is one level up from __tests__ directory
+
+  const { default: codemods } = await import('@ember-data/codemods');
+
+  if (!(transformName in codemods)) {
+    throw new Error('No codemod found for ' + transformName);
+  }
+
+  // FIXME: Gave up on types here
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
+  const transform: Transform = (codemods as any)[transformName];
+
+  const extension = extensionForParser(testOptions.parser);
+  const fixtureDir = path.join(dirName, '__testfixtures__');
+  const inputPath = path.join(fixtureDir, testFilePrefix + `.input.${extension}`);
+  const source = fs.readFileSync(inputPath, 'utf8');
+  const expectedOutput = fs.readFileSync(path.join(fixtureDir, testFilePrefix + `.output.${extension}`), 'utf8');
+  runInlineTest(
+    transform,
+    options ?? {},
+    {
+      path: inputPath,
+      source,
+    },
+    expectedOutput,
+    testOptions
+  );
+
+  let info = '{}';
+  try {
+    info = fs.readFileSync(path.join(fixtureDir, testFilePrefix + `.info.json`), 'utf8');
+  } catch {
+    /* empty */
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+  const expectedLogs = JSON.parse(info).expectedLogs ?? [];
+  expect(logs).toEqual(expectedLogs);
+
+  console.warn = realLoggerWarn;
+}
+
+// TODO: Fix js tests
+function extensionForParser(parser: TestOptions['parser']) {
+  switch (parser) {
+    case 'ts':
+    case 'tsx':
+      return parser;
+    default:
+      return 'js';
+  }
+}
+
+function runInlineTest(
+  module:
+    | {
+        default: Transform;
+        parser: TestOptions['parser'];
+      }
+    | Transform,
+  options: Options,
+  input: {
+    path?: string;
+    source: string;
+  },
+  expectedOutput: string,
+  testOptions?: TestOptions
+) {
+  const output = applyTransform(module, options, input, testOptions);
+  expect(output).toEqual(expectedOutput.trim());
+  return output;
+}
+
+// prettier-ignore
+runTests(
```

Comment:
jscodeshift's default testing setup is missing some features, so I wrote my own.

For each codemod, we can add a folder in `__testfixtures__` with the dasherized codemod name, and this test setup will run that codemod against all of the files contained therein. `testName.input.ts` will be modded and the result will be compared with `testName.output.ts`. You can also add `testName.info.json` with other useful stuff (e.g. config--not set up yet--or expected log output--see e.g. below).

example:
```
// testName.info.json
{
  "expectedLogs": [
    [
      "warn",
      "oh noes"
    ],
  ]
}

```